### DISCUSSION TITLE
Common positions pre-evaluated?

### DISCUSSION THREAD

**Discussion by DLSdufer** [official-stockfish/Stockfish#4286] [Ideas]
I have an idea. What if, some common positions are pre-evaluated on more powerful devices. This can also be used for openings, like it could have pre-evaluation at depth 99 for Giuoco piano, Accelerated London, Accelerated Dragon, Ruy Lopez, Smith-Morra Gambit, Open Sicilian, Closed Sicilian, Advanced Caro-Kann, Nimzo-Indian etcetera.

**Comment by XInTheDark** [comment]
That's called an opening book?

**Comment by bftjoe** [comment]
Depth 99 in an opening? Good luck calculating that...

**Comment by XInTheDark** [comment]
@TheDarkKnightRise There is a maximum depth of 245 though :)

**Comment by vondele** [comment]
There are engines (including SF forks) that can consult a database such as https://chessdb.cn/queryc_en/ or other opening books. This is not something we pursue for the Stockfish we build here.
